### PR TITLE
feature(ui5-application-writer): Add support for 'sapuxLayer' option

### PR DESCRIPTION
**Pull request by IainSAP**
Adds `sapuxLayer` as an option

### CODE REVIEW

File: packages/ui5-application-writer/src/index.ts [hunk 1]
```diff
@@ -28,7 +28,7 @@ async function generate(basePath: string, ui5App: Ui5App, fs?: Editor): Promise<
     // Add optional features
     if (ui5App.appOptions) {
         Object.entries(ui5App.appOptions).forEach(([key, value]) => {
-            if (value === true) {
+            if (value !== false) {
```
Comment (ullasholla) [inline packages/ui5-application-writer/src/index.ts:31]:
```suggestion
            if (value) {
```
Will a check for 'truthy' values suffice here?

File: packages/ui5-application-writer/test/options.test.ts [hunk 2]
```diff
@@ -11,9 +14,20 @@ describe('UI5 templates', () => {
         removeSync(outputDir); // even for in memory
     });
 
-    it('generates options', async () => {
-        const projectDir = join(outputDir, 'testapp_options');
-        const fs = await generate(projectDir, {
+    afterEach(() => {
```
Comment (ullasholla) [inline packages/ui5-application-writer/test/options.test.ts:17]: Nice. This is much cleaner.

File: packages/ui5-application-writer/test/options.test.ts [hunk 3]
```diff
@@ -11,9 +14,20 @@ describe('UI5 templates', () => {
...
+            // write out the files for debugging
+            if (debug) {
+                fs.commit(resolve);
+            } else {
+                resolve(true);
+            }
+        });
+    });
+
+    it('generates simple options', async () => {
```
Comment (ullasholla) [inline packages/ui5-application-writer/test/options.test.ts:28]: Do we mean 'with default' options?

Comment (IainSAP) [reply packages/ui5-application-writer/test/options.test.ts:28]: No there are no default options. There are simple types (boolean) which simply add some file entries. Then there are options that require template params.

File: packages/ui5-application-writer/src/index.ts (hunk 1 above)
Comment (IainSAP) [reply packages/ui5-application-writer/src/index.ts:31]: The condition is testing for cases where the value is of type boolean and is false explicitly.

Comment (ullasholla) [reply packages/ui5-application-writer/src/index.ts:31]: Is this the explanatory comment?
```
// Add optional features
```

So do we treat an option set to `undefined` or `''` as true?

File: packages/ui5-application-writer/test/options.test.ts (hunk 3 above)
Comment (IainSAP) [reply packages/ui5-application-writer/test/options.test.ts:28]: You could rightly ask what about writing a value thats a boolean...well that would be: 
```
appOptions: {
    someNewProp: 'true'
}
...
<%- someNewProp %>
...
```

**Pull request closed without merging**